src/ServicePulse.Host/vue/src/resources/Endpoint.ts

```diff
@@ -0,0 +1,96 @@
+export interface Endpoint {
+  name: string;
+  errorCount: number;
+  serviceControlId: string;
+  isScMonitoringDisconnected: boolean;
+  metrics: EndpointMetrics;
+  isStale: boolean;
+  endpointInstanceIds: string[];
+  disconnectedCount: number;
+  connectedCount: number;
+}
+
+export interface EndpointValues {
+  points: number[];
+  average: number;
+}
+
+export interface EndpointValuesWithTime extends EndpointValues {
+  timeAxisValues: Date[];
+}
+
+export interface EndpointMetrics {
+  [index: string]: EndpointValues;
+  queueLength: EndpointValues;
+  throughput: EndpointValues;
+  retries: EndpointValues;
+  processingTime: EndpointValuesWithTime;
+  criticalTime: EndpointValuesWithTime;
+}
+
+const defaultMetricData: EndpointValues = {
+  points: [],
+  average: 0,
+};
+const defaultTimeMetricData: EndpointValuesWithTime = {
+  ...defaultMetricData,
+  timeAxisValues: [],
+};
+export const emptyEndpointMetrics = (): EndpointMetrics => ({
+  queueLength: defaultMetricData,
+  throughput: defaultMetricData,
+  retries: defaultMetricData,
+  processingTime: defaultTimeMetricData,
+  criticalTime: defaultTimeMetricData,
+});
+
+export interface EndpointInstance {
+  name: string;
+  id: string;
+  isStale: boolean;
+  metrics: EndpointMetrics;
+}
+
+export interface ExtendedEndpointInstance extends EndpointInstance {
```

Comment:
Consider using a better name for this interface?

Reply:
i went through this with Mike - Extended seems like the best name, since these are values added onto the data retrieved from SC; literally extending the existing data

Reply:
To me, it seems this new type is to include data about the UI state, so I am wondering if it would be better to have a separate "state" type?